pkg/extensions/sync/sync.go

```diff
@@ -279,6 +281,30 @@ func getUpstreamContext(regCfg *RegistryConfig, credentials Credentials) *types.
 	return upstreamCtx
 }
 
+// getRepoDestination returns the local storage path of the synced repo based on the specified destination.
+func getRepoDestination(remoteRepo string, destination string) string {
+	if destination == "" {
+		return remoteRepo
+	}
+
+	remoteRepo = strings.Trim(remoteRepo, "/")
+	destination = strings.Trim(destination, "/")
+	remoteRepoSlice := strings.Split(remoteRepo, "/")
+	destinationSlice := strings.Split(destination, "/")
+
+	if len(destination) == 0 {
+		return remoteRepoSlice[len(remoteRepoSlice)-1]
+	}
+
+	if destinationSlice[len(destinationSlice)-1] == remoteRepoSlice[len(remoteRepoSlice)-1] {
+		return strings.Join(destinationSlice, "/")
```

Comment:
I don't think this is necessary, we use path.Join() everywhere so it should be ok.
Can you try without it?

Reply:
This was part of the old logic, I have updated the logic and came up with better ways to manipulate the strings